proxy/src/proxy.rs

```diff
@@ -101,6 +101,7 @@ async fn handshake<S: AsyncRead + AsyncWrite + Unpin>(
     stream: S,
     mut tls: Option<TlsConfig>,
     cancel_map: &CancelMap,
+    common_name: &Option<String>,
```

Comment:
`Option<&str>` is better than `&Option<String>`. You can't move the `String` using a reference anyway, and `None` is better than `&None`.

Reply:
ok, thanks